terraform/charm/sharded/variables.tf

```
@@ -0,0 +1,82 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+# variable "cluster_name" {
+#   type        = string
+#   description = "The cluster name of the fleet."
+#   default     = "opensearch"
+# }
+
+variable "config-server" {
+  description = "Config server orchestrator app definition"
+  type = object({
+    app_name          = string
+    model             = string
+    config            = optional(map(string), {role="shard"})
+    channel           = optional(string, "6/stable")
+    base              = optional(string, "ubuntu@22.04")
+    revision          = optional(string, 199)
```

Use the default value coming from the simple_deployment module.
Or (as I've seen it used as a string in the product module, update the variables.tf from simple deployment to set it as a string and not a number.
```suggestion
    revision          = optional(number, null)
```